src/arch/riscv/tlb.cc

```diff
@@ -385,9 +385,16 @@ TLB::translate(const RequestPtr &req, ThreadContext *tc,
         if (fault == NoFault) {
             if (req->getFlags() & Request::PHYSICAL) {
                 /**
-                 * we simply set the virtual address to physical address
+                 * we simply set the virtual address to physical address.
+                 *
+                 * For RV32, we follow what the specification said:
+                 * When mapping between narrower and wider addresses,
+                 * RISC-V zero-extends a narrower physical address to a
+                 * wider size.
                  */
-                req->setPaddr(req->getVaddr());
+                req->setPaddr(((ISA*) tc->getIsaPtr())->rvType() == RV32 ?
+                              req->getVaddr() & 0xffffffffLL :
```

Comment:
If we're going to do this, let's create a function in case there are other places where the logic needs to be replicated. Also, please use `bits(31,0)` or something similar instead of manually masking.

Reply:
Fixed.